exp/tools/serpapi/internal/serpapi_client.go

```diff
@@ -0,0 +1,109 @@
+package internal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+var ErrMissingToken = errors.New("missing the OpenAI API key, set it in the SERPAPI_API_KEY environment variable")
+var NoGoodResult = "No good search result found"
```

Comment:
I am not sure about this, but in my experience, users cannot import the variables, types, functions, and packages from `internal` package. What will happen if someone tries to do so?